test/unit/lib/getMergedPullRequestsSpec.js

```diff
@@ -120,4 +120,18 @@ describe('getMergedPullRequests', function () {
         return expect(getMergedPullRequests(anyRepo))
             .to.become(expectedResults);
     });
+
+    it('should skip with non-matching parenthesis', function () {
+        const gitLogMessages = [
+            'Merge pull request #3 from kadirahq/greenkeeper-update-all'
```

Comment:
Agreed, the refactoring is really nice. I’m still not sure if this already fixes the problem, so please let me know if it doesn’t fixes the problem.